### DISCUSSION TITLE
最低 Node.js 版本支持更新到 14.x

### DISCUSSION THREAD

**Discussion by Aaaaash** [opensumi/core#1010] [RFC]
## 背景
1. Node.js 12.x 已经进入维护期
2. 越来越多的 VS Code 插件已经开始使用 ES2020 的 optional operator 语法

## 需要修改的点
1. tsconfig & ts-loader 中的 `target` 更新为 `es2020`
2. 部分 webpack 依赖升级，已知支持 es2020 的有
    1. terser-webpack-plugin  4.x
    2. terser 5.x
    3. acorn 8.x

**Comment by erha19** [comment]
NodeJS 14.5.0 版本开始支持大部分（99%） es2020 特性，见：[node.green](https://node.green/), 关于 es2020 的一些重要特性：

1.  [javascript-whats-new-es2020/](https://auth0.com/blog/javascript-whats-new-es2020/)
2. [ecmascript-2020-biggest-new-features](https://areknawo.com/ecmascript-2020-biggest-new-features/)
